dwl.c

```diff
@@ -1212,8 +1214,19 @@ renderindependents(struct wlr_output *output, struct timespec *now)
 		rdata.x = c->surface.xwayland->x;
 		rdata.y = c->surface.xwayland->y;
 
+		if (c->surface.xwayland->parent) {
+			pg.x = c->surface.xwayland->parent->x;
+			pg.y = c->surface.xwayland->parent->y;
+			pg.width = c->surface.xwayland->parent->width;
+			pg.height = c->surface.xwayland->parent->height;
+			wlr_renderer_scissor(drw, &pg);
```

Comment:
Firefox and IntelliJ menus follow this branch and are indeed restricted to the parent window.

Chromium menus and IntelliJ windows follow the other branch and are free to render outside the parent.